src/Prefixer.js

```diff
@@ -59,6 +66,15 @@ export function prefix (value, length) {
 		// justify-content
 		case 4968:
 			return replace(replace(value, /(.+:)(flex-)?(.*)/, WEBKIT + 'box-pack:$3' + MS + 'flex-pack:$3'), /s.+-b[^;]+/, 'justify') + WEBKIT + value + value
+		// justify-self
+		case 4200:
+			// center, end, start, stretch
+			if (
+				match(substr(value, length + 1, sizeof(value) - 1), /^(center|end|start|stretch)$/)
```

Comment:
> Why not just match on the value i.e `match(value, /(cen|end|sta|str).*/)` ?

The value must be matched exactly, preventing `flex-(start|end)`, `self-(start|end)` from being prefixed.

Reply:
and the matching on value instead of substr(value, length + 1, sizeof(value - 1)) part, you're extracting the prop from `key:prop;` but since we are sure that the key cannot possibly match the list of options we don't need to exclude the key from the match? That is `center|end|start|stretch` cannot possibly be in `justify-self:`.

Reply:
> and the matching on value instead of substr(value, length + 1, sizeof(value - 1)) part, you're extracting the prop from `key:prop;` but since we are sure that the key cannot possibly match the list of options we don't need to exclude the key from the match? That is `center|end|start|stretch` cannot possibly be in `justify-self:`.

But `start|end` can possibly be in `(flex|self)-(start|end)`. To prevent `(flex|self)-(start|end)` from being matched, it has to match the value exactly, or at least match from the start of the value. Both require strip `key` from `key:prop;`.

Reply:
I meant the first part namely why `match(substr(value, length + 1, sizeof(value) - 1, ...)` and not just `match(value, ...)`

Reply:
@thysultan Done in https://github.com/thysultan/stylis/pull/285/commits/999c88505871ef1e0ebc5d729976ab28e7b096db. Note checking the absence of `flex-` is still required.